core/BUILD

```diff
@@ -2,8 +2,9 @@
 # Exceptions. See /LICENSE for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+# The prelude, and all of its dependencies.
 filegroup(
-    name = "core",
-    data = glob(["**/*.carbon"]),
+    name = "prelude",
+    data = ["prelude.carbon"] + glob(["prelude/**/*.carbon"]),
```

Comment:
What model do you expect when we start having more files in core than just the prelude? Shouldn't the driver and tests have those available too?

Reply:
My expectation is that toolchain tests should not depend on any part of core other than the prelude. I think tests for the rest of core should live in `//core` instead, treating the rest of core as a layer on top of the base toolchain.

For `carbon_binary` or `carbon_library` targets, I'm expecting we would take a similar stance: each target should be explicitly declaring a dependency on any libraries they use (including core libraries). If that turns out to be excessively verbose or non-ergonomic, we could inject an implicit dependency on `//core` into each target, but to the extent possible, I think we should strive to treat core as "just another package" to the extent we can. The prelude is an exception to this because we implicitly import it (but even then, I think we should be looking at ways we can make it less special in the longer term).

Reply:
To try to wrap back from discussion yesterday -- I think when we run the driver through bazel, or when we install, we want to be pulling in _all_ core files as dependencies, not just the prelude.

Carbon's toolchain tests are going to be difficult to split up. We're running into linking size issues, and that's going to be a significant limiting factor for splitting things up.

With regards to the ability to set narrow dependencies, I don't think we're there right now. Also, for a small project, I would be concerned that having to individually set dependencies _in addition to_ individually `import`ing dependencies is going to be regarded as duplicative.